### PR TITLE
upgrade to controller-runtime 0.7.2

### PR DESCRIPTION
**Pull request by joelddiaz**
vendor in controller-runtime 0.7.2

update affected code

 Package location changes:
    change in signal package from pkg/runtime/signals -> pkg/manager/signals
    
Controller updates:
    update controller predicates from e.Meta -> e.Object
    update EnqueueRequestsFromMapFunc to new style registration
    update Reconcile() function signature to accept context.Context as first
    parameter
    update test cases to pass dummy context.Context

This PR builds on the changes found in #407

### CODE REVIEW

File: pkg/operator/cleanup/cleanup_controller_test.go [hunk 1]
```diff
@@ -123,7 +123,8 @@ func TestStaleCredentialsRequestReconcile(t *testing.T) {
 				Client: fakeClient,
 			}
 
-			_, err := rscr.Reconcile(reconcile.Request{
+			ctx := context.TODO()
+			_, err := rscr.Reconcile(ctx, reconcile.Request{
```
Comment (2uasimojo) [inline pkg/operator/cleanup/cleanup_controller_test.go:127]: Any reason `context.TODO()` isn't inlined here, like "normal"?

**Comment by 2uasimojo** [comment]
Started reviewing this before realizing it's stacked on #407. Can come back to it once that's merged and this is rebased.

File: pkg/operator/cleanup/cleanup_controller_test.go (hunk 1 above)
Comment (joelddiaz) [reply pkg/operator/cleanup/cleanup_controller_test.go]: No particular reason. Fixed and pushed.

File: pkg/operator/secretannotator/aws/reconciler_test.go [hunk 2]
```diff
@@ -277,7 +277,8 @@ func TestSecretAnnotatorReconcile(t *testing.T) {
 				},
 			}
 
-			_, err := rcc.Reconcile(reconcile.Request{
+			ctx := context.TODO()
+			_, err := rcc.Reconcile(ctx, reconcile.Request{
```
Comment (2uasimojo) [inline pkg/operator/secretannotator/aws/reconciler_test.go:281]: .

File: pkg/operator/status/status_controller_test.go [hunk 3]
```diff
@@ -71,7 +71,8 @@ func TestClusterOperatorVersion(t *testing.T) {
 				Logger:   log.WithField("controller", "teststatus"),
 				platform: configv1.AWSPlatformType,
 			}
-			_, err := rs.Reconcile(reconcile.Request{})
+			ctx := context.TODO()
+			_, err := rs.Reconcile(ctx, reconcile.Request{})
```
Comment (2uasimojo) [inline pkg/operator/status/status_controller_test.go:75]: .

File: pkg/operator/status/status_controller_test.go [hunk 4]
```diff
@@ -310,7 +311,8 @@ func TestConditions(t *testing.T) {
 		}
 		defer clearHandlers()
 
-		_, err := r.Reconcile(reconcile.Request{})
+		ctx := context.TODO()
+		_, err := r.Reconcile(ctx, reconcile.Request{})
```
Comment (2uasimojo) [inline pkg/operator/status/status_controller_test.go:315]: .

**Review by 2uasimojo — COMMENTED** [review]
Reviewed this minus the first commit. Seems straightforward. Assuming CI is happy, this should be good to go once the predecessor lands.

Noted a couple more spots where context could be inlined, but that's totally not important to fix.

**Comment by joelddiaz** [comment]
> Reviewed this minus the first commit. Seems straightforward. Assuming CI is happy, this should be good to go once the predecessor lands.
> 
> Noted a couple more spots where context could be inlined, but that's totally not important to fix.

No reason not to get some consistency. It'll be a while before the dependent #407 PR merges

**Comment by 2uasimojo** [comment]
/lgtm

Unhold at will.

**Comment by joelddiaz** [comment]
/assign @jeana-redhat @lwan-wanglin @sferich888 
This should have no visible impact. Just trying to not fall too behind on our controller-runtime dependency.

**Comment by jeana-redhat** [comment]
/label docs-approved
(no docs)

**Comment by lwan-wanglin** [comment]
Sorry for missing this one, did some work, no regression until now, test run: [OS-20211118-0634](https://polarion.engineering.redhat.com/polarion/#/project/OSE/testrun?id=OS-20211118-0634)
test job: https://mastern-jenkins-csb-openshift-qe.apps.ocp-c1.prod.psi.redhat.com/job/ocp-common/job/Runner/224986/consoleFull

/label qe-approved

**Comment by joelddiaz** [comment]
@sferich888 px-approved label?

**Pull request merged**